repo/datastore.go

```diff
@@ -341,3 +373,28 @@ type ModeratedStores interface {
 	// Delete a moderated store from the database
 	Delete(peerId string) error
 }
+
+type KeyStore interface {
+	Queryable
+	wallet.Keys
+}
+
+type SpentTransactionOutputStore interface {
+	Queryable
+	wallet.Stxos
+}
+
+type TransactionStore interface {
+	Queryable
+	wallet.Txns
+}
+
+type UnspentTransactionOutputStore interface {
+	Queryable
+	wallet.Utxos
+}
+
+type WatchedScriptStore interface {
+	Queryable
+	wallet.WatchedScripts
+}
```

Comment:
This will let us abstract our Datastore internals from the wallet interface... making it easier to decouple these packages later.